### DISCUSSION TITLE
default.Event.GET-positive.EventGET001  is failing due to variable count in each test case execution

### DISCUSSION THREAD

**Discussion by divyashree123456** [edgexfoundry/community#153] [Q&A]
Hi @cloudxxx8 @lenny-intel ,
We are Running test cases using our own images on riscv machine using docker.[ Jakarta version ]
**GET-positive.EventGET001- Query all events**

![Screenshot from 2023-07-24 18-15-32](https://github.com/edgexfoundry/community/assets/120623642/4ad18dfd-9b18-4e1a-855a-2b157852e31e)

1.It is failing in the highlighted lines as shown in image, Similar test cases are failing with count != specified integer ,and sometimes its passing the test case. why there is variation in the count ,because sometimes I m getting count equal to 7 or 9 or 11 which not equal to specified value.
2. And why it is specified as **totalCount Should be 6** in default.Event.GET-positive.EventGET001 test case.

![Screenshot from 2023-07-24 17-55-08](https://github.com/edgexfoundry/community/assets/120623642/779a8ce8-818b-4b2b-8a23-36d025d575b4)

please help me to understand this test case.
Thanks in advance

**Comment by cloudxxx8** [comment]
The robot file is quite easy to understand.
Create 6 events and verify there are 6 evetns received.
The log shows you got 10 events.  You might check why you got additional events during the tests.  I believe it's not risc-v related.

**Comment by divyashree123456** [comment]
Hi @cloudxxx8 ,Can you please tell me ,where I can create these 6 events ?

**Reply by cloudxxx8** [reply]
Please check the robot framework keyworkd in the robot file.
In this case, it's `Create multiple Events` in the `Given` section.

**Reply by divyashree123456** [reply]
Hi @cloudxxx8 ,can we create any number of devices?Like as in this script defining 6 events ,can we create 8 events ?

**Reply by cloudxxx8** [reply]
you can modify the test script to meet your needs

**Comment by divyashree123456** [comment]
Hi @cloudxxx8  ,One more clarification for running a Test cases of **CORE-DATA**  is it necessary to run device-virtual service?
Because earlier I was facing the inconsistency while testing the test cases ,sometimes there were failing or passing.but when I tried running test-cases for **core-data ,by stoping device-virtual containe**r ,I was getting constitent report of test cases execution.
